### PR TITLE
Persistence

### PR DESCRIPTION
**Pull request by foxtree**
on top of #173

### CODE REVIEW

File: src/redux/action-types/index.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export const actionTypes = { // eslint-disable-line import/prefer-default-export
   ERROR: 'ERROR',
   SET_AUTH: 'auth/setAuth', // implicitly derived from src/features/auth/authSlice.js
+  // REHYDRATE: 'persist/REHYDRATE',
```
Comment (gilhsu) [inline src/redux/action-types/index.js]: do we need this?

File: src/redux/index.js [hunk 2]
```diff
@@ -12,23 +17,31 @@ import {
...
     collections: collectionsReducer,
   });
 
+  const persistReducerConfig = {
+    version: '0.7.4',
+    key: `patient-${patientId}`,
+    // key: `root:patient:${patientId}`,
+    storage: AsyncStorage,
+    whitelist: ['activeCollectionId', 'collections'],
+    // stateReconciler: hardSet,
```
Comment (gilhsu) [inline src/redux/index.js]: cleanup

File: src/redux/index.js [hunk 3]
```diff
@@ -41,7 +54,28 @@ const createStore = () => {
...
+  const persistor = persistStore(store, persistConfig, callback);
+  // persistor.purge();
+  //     purges state from disk and returns a promise
+  // persistor.flush();
+  //     immediately writes all pending state to disk and returns a promise
+  // persistor.pause();
+  //     pauses persistence
+  // persistor.persist();
+  //     resumes persistence
+
```
Comment (gilhsu) [inline src/redux/index.js:60]: do we need this?

File: src/redux/reducers/index.js [hunk 4]
```diff
@@ -146,9 +146,11 @@ const createNote = (text) => {
 export const collectionsReducer = (state = preloadCollections, action) => {
   switch (action.type) {
     case actionTypes.CLEAR_PATIENT_DATA: {
-      const newDefaultCollection = createCollection();
+      const { id } = defaultCollection;
+      defaultCollection = createCollection();
```
Comment (gilhsu) [inline src/redux/reducers/index.js:150]: how is this not a name clash?

File: src/redux/reducers/index.js [hunk 5]
```diff
@@ -146,9 +146,11 @@ const createNote = (text) => {
...
+      defaultCollection.id = id;
       return {
-        [newDefaultCollection.id]: newDefaultCollection,
+        [defaultCollection.id]: defaultCollection,
```
Comment (gilhsu) [inline src/redux/reducers/index.js]: These parts of redux probably get refactored once we pause and purge the store on logout. There shouldn't be any collections data after logging out.

Comment (gilhsu) [reply src/redux/reducers/index.js]: TODO for the next PR relating to pause and purge

**Review by gilhsu — APPROVED** [review]

File: src/redux/index.js (hunk 3 above)
Comment (foxtree) [reply src/redux/index.js:60]: inline documentation

Comment (foxtree) [reply src/redux/index.js:60]: removed.

**Pull request merged**